go/vt/vttablet/tabletserver/vstreamer/snapshot_conn.go

```diff
@@ -217,3 +217,43 @@ func GetBinlogRotationThreshold() int64 {
 func SetBinlogRotationThreshold(threshold int64) {
 	atomic.StoreInt64(&binlogRotationThreshold, threshold)
 }
+
+func (conn *snapshotConn) startSnapshotAllTables(ctx context.Context) (gtid string, err error) {
+	lockConn, err := mysqlConnect(ctx, conn.cp)
```

Comment:
What is the context here? We may want to create a child context with a short timeout as we don't want to be holding the global read lock for long.

Reply:
We can potentially add a `startSnapshotAllTablesWithConsistentGTID()` function, in the likeness of `startSnapshotWithConsistentGTID()`? Which avoids the need to issue a `LOCK` or a `FLUSH TABLES` in the first place.

Reply:
Made the change to use startSnapshotWithConsistentGTID(). Can you check if has been done right? I didn't see  that startSnapshotWithConsistentGTID() was scoped to a single table.

Reply:
I don't see any further change in `snapshot_conn.go`? Did you commit & push that change?

Reply:
Ah, found it in [tablestreamer.go](https://github.com/vitessio/vitess/pull/13137#diff-b5dcae405a83e01e33acc61a77c9254d0c52498a83bb2e4cca01c593d06cb90f). The force-push confused me. Yes, it looks right!